apps/fly-my-rockets/src/app/rockets/rocket.service.ts

```diff
@@ -32,12 +32,21 @@ export class RocketService {
     return from(this.db.collection('rockets').doc(rocketId).delete());
   }
 
-  updateFlights(rocketId: string, flightIds: string[]) {
-    flightIds = flightIds.map(fid => `/flights/${fid}`);
-    return from(this.db.collection('rockets').doc(rocketId).update({ flights: FieldValue.arrayUnion(...flightIds) }));
+  updateFlight(rocketId: string, oldFlight: Flight, updatedFlight: Flight): Observable<any> {
+    const rocketDoc = this.db.firestore.collection('rockets').doc(rocketId);
+    return from(this.db.firestore.runTransaction(transaction => {
+      return transaction.get(rocketDoc).then(doc => {
+        transaction.update(rocketDoc, {
+          flights: FieldValue.arrayRemove(oldFlight)
+        });
+        transaction.update(rocketDoc, {
+          flights: FieldValue.arrayUnion(updatedFlight)
+        });
```

Comment:
To 'update' a flight, we need to remove the old one from the array and add the updated one. Do this in a transaction so we don't lose a flight on delete if the add is unsuccessful.